tests/buffer/associative_containers.cpp

```diff
@@ -10,7 +10,7 @@
 
 using namespace cl::sycl;
 
-int test_main(int argc, char *argv[]) {
+int test_main(int /*argc*/, char** /**argv[]*/) {
```

Comment:
I find the previous type of `arg` clearer, since it correctly suggest it is an array of C-style strings.
Every time I use a pointer in my C++ programs, it is a kind of personal failure. So if I can minimize the number of `*` I am happier. :-)

Is there a strong compelling argument for these changes?